docs/rfc/block-registration.md

```diff
@@ -0,0 +1,414 @@
+		{ "name": "default", "label": "Default", "isDefault": true }, 
+		{ "name": "other", "label": "Other" }
+	]
+}
+```
+
+Block styles can be used to provide alternative styles to block. It works by adding a class name to the block's wrapper. Using CSS, a theme developer can target the class name for the style variation if it is selected.
+
+Plugins and Themes can also register [custom block style](https://wordpress.org/gutenberg/handbook/designers-developers/developers/filters/block-filters/#block-style-variations) for existing blocks.
+
+### Transforms
+
+* Type: `string`
+* Optional
+* Localized: No
+* Property: `transforms`
+
+```json
+{ "transforms": "my-block-transforms.js" }
+```
+
+This property is a pointer to a JavaScript file containing the save function of the block transforms. The save function defines the way in which the different attributes should be combined into the final markup, which is then serialized by Gutenberg into `post_content`.
+
+See the [Transforms](https://wordpress.org/gutenberg/handbook/designers-developers/developers/block-api/block-registration/#transforms-optional) documentation for more details.
+
+### Deprecated versions
+
+* Type: `object[]`
+* Optional
+* Localized: No
+* Property: `deprecated`
+
+```json
+{ "deprecated": [ {
+	"attributes": {},
+	"save": "my-deprecated-save.js",
+	"supports": {}
+} ] }
+```
+
+This property contains the definition of the deprecated versions of the block type. It is used to ensure that old blocks with old markup are not considered invalid.
+
+See the [Deprecated Blocks](https://wordpress.org/gutenberg/handbook/designers-developers/developers/block-api/block-deprecation/) documentation for more details.
+
+### Supports
+
+* Type: `object`
+* Optional
+* Localized: No
+* Property: `supports`
+
+Optional block extended support features. 
+
+See the [block supports](https://wordpress.org/gutenberg/handbook/designers-developers/developers/block-api/block-registration/#supports-optional) documentation page for more details.
+
+### Stylesheets
+
+* Type: `object`
+* Optional
+* Localized: No
+* Property: `stylesheets`
+
+```json
+{
+	"stylesheets": {
+		"main": "my-block-style.css",
+		"editor": "my-block-editor-style.css",
+		"theme": "my-block-theme-style.css"
+	}
+}
+```
+
+This property is a pointer to CSS files containing the CSS used for the block in different contexts.
+
+## Internationalization
+
+Localized properties are automatically wrapped in `_x` function calls on the backend and the frontend of WordPress. These translations are added as an inline script to the `wp-block-library` script handle in WordPress core or to the plugin's script handle when it defines metadata defintion.
+
+WordPress string discovery automatically includes these strings to the plugin's or core's domain name.
+
+**Example:**
+
+```json
+{
+	"title": "My block",
+	"description": "My block is fantastic",
+	"keywords": [ "fanstastic" ]
+}
+```
+
+in JavScript with help of Babel plugin becomes:
```

Comment:
Typo: "JavaScript"

Seems a full sentence, so capitalize appropriate?

It's probably not obvious to the reader what you're referring to with the Babel plugin. A link to a related resource could help.

```suggestion
In JavaScript, with the help of a Babel plugin, this becomes:
```